File: Leetcode-Python/data_structure/tree_node.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

    def array_to_tree(self, arr, root, i):
        if i < len(arr):
            temp = TreeNode(arr[i])
            root = temp

            root.left = self.array_to_tree(arr, root.left, 2 * i + 1)
            root.right = self.array_to_tree(arr, root.right, 2 * i + 2)

        return root

    def print_tree_in_order(self, root):
        if root:
            self.print_tree_in_order(root.left)
            print(root.val)
            self.print_tree_in_order(root.right)

File: Leetcode/Java/src/Fundamentals/DataStructures/TreeNode.java
package Fundamentals.DataStructures;

public class TreeNode {

    public int val;
    public TreeNode left;
    public TreeNode right;

    public TreeNode() {

    }

    public TreeNode(int x) {
        val = x;
    }

    public static TreeNode getBalancedTree() {
        // Level 0
        TreeNode root = new TreeNode(1);

        // Level 1
        root.left = new TreeNode(2);
        root.right = new TreeNode(3);

        // Level 2
        root.left.left = new TreeNode(4);
        root.left.right = new TreeNode(5);
        root.right.left = new TreeNode(6);
        root.right.right = new TreeNode(7);

        return root;
    }

    public static TreeNode getLinkedList() {
        TreeNode root = new TreeNode(1);

        root.left = new TreeNode(2);
        root.left.left = new TreeNode(3);

        root.left.left.left = new TreeNode(4);
        root.left.left.left.left = new TreeNode(5);
        root.left.left.left.left.left = new TreeNode(6);
        root.left.left.left.left.left.left = new TreeNode(7);

        return root;
    }

    public static TreeNode getEmptyTree() {
        return null;
    }
}
